src/stability_sdk/animation.py

```diff
@@ -780,24 +787,34 @@ def _span_render_frame(self, frame_idx: int, init: Optional[np.ndarray], mask: O
         )
         return results[generation.ARTIFACT_IMAGE][0]
 
-    def _span_render(self, start: int, end: int, prev_frame: Optional[np.ndarray]) -> Generator[Tuple[int, np.ndarray], None, None]:
+    def _span_render(self, start: int, end: int, seed: int, prev_frame: Optional[np.ndarray]) -> Generator[Tuple[int, np.ndarray], None, None]:
         args = self.args
 
-        def apply_xform(frame: np.ndarray, xform: matrix.Matrix) -> Tuple[np.ndarray, np.ndarray]:
+        def apply_xform(frame: np.ndarray, xform: matrix.Matrix, frame_idx: int) -> Tuple[np.ndarray, np.ndarray]:
+            args, frame_args = self.args, self.frame_args
             if args.animation_mode == '2D':
                 xform = to_3x3(xform)
-            frames, masks = self.api.transform([frame], resample_op(args.border, xform, export_mask=True))
+                frames, masks = self.api.transform([frame], resample_op(args.border, xform, export_mask=True))
+            else:
+                fov = frame_args.fov_series[frame_idx]
+                depth_blur = int(frame_args.depth_blur_series[frame_idx])
+                depth_warp = frame_args.depth_warp_series[frame_idx]
+                projection = matrix.projection_fov(math.radians(fov), 1.0, args.near_plane, args.far_plane)                
+                wvp = matrix.multiply(projection, xform)
+                depth_calc = depthcalc_op(args.depth_model_weight, depth_blur)
+                resample = resample_op(args.border, wvp, projection, depth_warp=depth_warp, export_mask=True)
+                frames, masks = self.api.transform_resample_3d([frame], depth_calc, resample)
```

Comment:
It seems that i've missing it, what it is the workaround for 3D case? This implementation seems straightforward (I've expected some trick for the 3D transform inversion)

Reply:
Oh the code I already had covers both 2D and 3D transform accumulation and inversion. The `build_frame_xform` method returns 4x4 matrix handling both 2D and 3D.

On the forward pass: https://github.com/Stability-AI/stability-sdk/blob/pharma.anim-updates/src/stability_sdk/animation.py#L816
```
            accum_xform = matrix.multiply(self.build_frame_xform(frame_idx), accum_xform)
```

On the backward pass: https://github.com/Stability-AI/stability-sdk/blob/pharma.anim-updates/src/stability_sdk/animation.py#L836
```
            frame_xform = self.build_frame_xform(frame_idx+1)
            inv_xform = np.linalg.inv(frame_xform).tolist()
            accum_xform = matrix.multiply(inv_xform, accum_xform)
```

So the only piece that was missing which this change adds is calling `transform_resample_3d` in the 3D case during application of the transform. This way the depth inference and camera projection matrix also get used.